compiler/shared_ast/dune

```diff
@@ -0,0 +1,6 @@
+(library
+ (name shared_ast)
+ (public_name catala.shared_ast)
+ (flags
+  (:standard -short-paths))
```

Comment:
What's that?

Reply:
It tells the OCaml compiler to choose the shortest string when printing types and values. Normally, I don't use it, but `dune` uses tricks to namespace libraries that are not so well hidden behind the scenes, and here with the "explicit toplevel module" library scheme (library `shared_ast` with a `shared_ast.ml` file that defines what to export), messages would become unreadable with `Shared_ast__.Types.` prefixes everywhere.

This is not a real fix (if only namespaces were well integrated into OCaml :roll_eyes: ) but it makes messages more readable. No effect on compilation otherwise.

Reply:
Wouahou that's very neat.